src/components/application_manager/include/application_manager/commands/command_request_impl.h

```diff
@@ -112,8 +112,15 @@ class CommandRequestImpl : public CommandImpl,
  public:
   enum RequestState { kAwaitingHMIResponse = 0, kTimedOut, kCompleted };
 
+  /**
+   * @brief The HashUpdateMode enum defines whether request has to update
+   * hash after its execution is finished
+   */
+  enum HashUpdateMode { kDoHashUpdate, kNoHashUpdate };
+
   CommandRequestImpl(const MessageSharedPtr& message,
-                     ApplicationManager& application_manager);
+                     ApplicationManager& application_manager,
+                     HashUpdateMode hash_update_mode = kNoHashUpdate);
```

Comment:
@AKalinich-Luxoft 
We agreed with A. Kutsan to move hash mode set-up into Init() function. This method is called for every command before Run().